internal/platform/platform.go

```diff
@@ -132,24 +148,32 @@ func (pi *PlatformInfo) listDpuDevices() ([]ghw.PCIDevice, []VendorDetector, err
 	return dpuDevices, activeDetectors, nil
 }
 
+func (pi *PlatformInfo) detectDpuSystem() (VendorDetector, error) {
+	dpuDevices, detectors, err := pi.listDpuDevices()
+	if err != nil {
+		return nil, errors.Errorf("Failed to get VspPlugin from platform: %v", err)
+	}
+	if len(dpuDevices) == 0 {
+		return nil, fmt.Errorf("Failed to detect any DPU devices")
+	}
+	if len(dpuDevices) != 1 {
+		return nil, fmt.Errorf("%v DPU devices detected. Currently only supporting exactly 1 DPU per node", len(dpuDevices))
+	}
+
+	return detectors[0], nil
+}
+
 func (pi *PlatformInfo) VspPlugin(dpuMode bool, vspImages map[string]string, client client.Client) (*plugin.GrpcPlugin, error) {
 	var detector VendorDetector
+	var err error
+
 	if dpuMode {
-		// TODO: We need to also detect the platform in dpuMode
-		detector = NewIntelDetector()
+		detector, err = pi.detectDpuPlatform()
```

Comment:
I don't think this code path has been exercised before for the IPU, we should make sure this is green in the e2e CI run.

Reply:
does the github action on this PR contains such a test?

Reply:
Yes, this will be tested by ci/prow/e2e-dpu-operator-tests. I don't see any reason this wouldn't work, but since we have hardcoded this in the past we should validate this is green on both Intel and Marvell hardware